src/components/analyticVisualization/EditVisualization.js

```diff
@@ -0,0 +1,96 @@
+import React, {useEffect, useState} from 'react'
+import i18n from '@dhis2/d2-i18n'
+import PropTypes from 'prop-types'
+import {
+    Button,
+    ButtonStrip,
+    Modal,
+    ModalActions,
+    ModalContent,
+    ModalTitle,
+} from '@dhis2/ui'
+import {NotEditableElement, VisualizationGroup, VisualizationTitle} from "./index";
+
+const EditVisualization = ({open, settings, handleChange, currentGroup, groups, handleEdit, handleClose, type}) => {
+    const [typeName, useTypeName] = useState('')
+    const [label, useLabel] = useState('')
+
+    const chooseTypeName = type => {
+        switch (type) {
```

Comment:
NIT: use a [lookup table pattern](https://omwri.medium.com/javascript-patterns-lookup-tables-26bbaf693e24) instead of the `switch` statement.

Reply:
Hi, thanks for the suggestion. I'm changing it to:

```
const typeName = typeNameList[type] || ''
const label = labelList[type] || ''
```